nip47/notifications/nip47_notifier.go

```diff
@@ -98,38 +99,40 @@ func (notifier *Nip47Notifier) notifySubscribers(ctx context.Context, notificati
 		if !hasPermission {
 			continue
 		}
-		notifier.notifySubscriber(ctx, &app, notification, tags)
-	}
-}
-
-func (notifier *Nip47Notifier) notifySubscriber(ctx context.Context, app *db.App, notification *Notification, tags nostr.Tags) {
-	logger.Logger.WithFields(logrus.Fields{
-		"notification": notification,
-		"appId":        app.ID,
-	}).Debug("Notifying subscriber")
 
-	var err error
+		appWalletPrivKey := notifier.keys.GetNostrSecretKey()
+		if app.WalletPubkey != nil {
+			appWalletPrivKey, err = notifier.keys.GetAppWalletKey(app.ID)
+			if err != nil {
+				logger.Logger.WithFields(logrus.Fields{
+					"notification": notification,
+					"appId":        app.ID,
+				}).WithError(err).Error("error deriving child key")
+				return
+			}
+		}
 
-	appWalletPrivKey := notifier.keys.GetNostrSecretKey()
-	if app.WalletPubkey != nil {
-		appWalletPrivKey, err = notifier.keys.GetAppWalletKey(app.ID)
+		appWalletPubKey, err := nostr.GetPublicKey(appWalletPrivKey)
 		if err != nil {
 			logger.Logger.WithFields(logrus.Fields{
 				"notification": notification,
 				"appId":        app.ID,
-			}).WithError(err).Error("error deriving child key")
+			}).WithError(err).Error("Failed to calculate app wallet pub key")
 			return
 		}
-	}
 
-	ss, err := nip04.ComputeSharedSecret(app.AppPubkey, appWalletPrivKey)
-	if err != nil {
-		logger.Logger.WithFields(logrus.Fields{
-			"notification": notification,
-			"appId":        app.ID,
-		}).WithError(err).Error("Failed to compute shared secret")
-		return
+		go notifier.notifySubscriber(ctx, &app, notification, tags, appWalletPubKey, appWalletPrivKey, false)
+		go notifier.notifySubscriber(ctx, &app, notification, tags, appWalletPubKey, appWalletPrivKey, true)
 	}
+}
+
+func (notifier *Nip47Notifier) notifySubscriber(ctx context.Context, app *db.App, notification *Notification, tags nostr.Tags, appWalletPubKey, appWalletPrivKey string, legacy bool) {
```

Comment:
rather than passing `legacy bool` we could pass the version number, and then use this abstraction idea I mentioned to remove the large conditional.